learn/guides/calling-java-code-from-ballerina.md

```diff
@@ -0,0 +1,290 @@
+	org.yaml.snakeyaml.introspector.BeanAccess
+	java.util.function.BiFunction
+	org.yaml.snakeyaml.DumperOptions$FlowStyle
+	...
+	...
+```
+
+- The `-mvn` option specifies the Maven dependency of the Java library required to generate bindings.
+- The argument list specifies the Java class names. 
+
+The `bindgen` tool generates bindings for:
+- The specified Java classes.
+- The Java classes exposed in the public APIs of all the specified classes.
+
+
+Before we move onto the next step, let’s verify the generated code. 
+```sh
+> bal build
+... 
+...
+
+Generating executable
+	target/bin/yaml_package.jar
+
+> bal run target/bin/yaml_package.jar
+Hello World!
+```
+
+### Step 4 - Writing the Ballerina Code
+Now, we’ll use the generated bindings and write the Ballerina code, which uses the SnakeYAML library. Here is the Java code. Let’s develop the corresponding Ballerina code step by step. 
+```java
+public class SnakeYamlSample {
+
+    public static void main(String... a) {
+        String filename = a[0];
+        try (InputStream inputStream = new FileInputStream(filename)) {
+            Yaml yaml = new Yaml();
+            Map<String, Object> obj = yaml.load(inputStream);
+            System.out.println(obj);
+        } catch (Exception e) {
+            System.err.println("The file '" + filename + "' cannot be loaded. Reason: " + e.getMessage());;
+        }
+    }
+}
+```
+
+#### Creating the `FileInputStream`
+Our goal here is to create a new `java.io.FileInputStream` instance from the filename. In Step 3, we generated bindings for the required Java classes. The following is the code snippet that does the job.
+
+```ballerina
+javaio:FileInputStream | javaio:FileNotFoundException fileInputStream = javaio:newFileInputStream3(filename);
+```
+
+Here, `FileInputStream` is the Ballerina class generated for the `java.io.FileInputStream` class.
+- Ballerina bindings for each Java package are mapped onto a separate Ballerina module by default. Therefore, you need to import them when using them inside other modules. Here the `java.io` Ballerina module (mapping the corresponding Java package) is imported as `javaio`. However, if you wish to generate all the bindings inside a single directory you can do so by using the `[(-o|--output) <output-path>]` command option.
+- You can find functions that start with `newFileInputStream` in the generated code. Each such function creates a new `java.io.FileInputStream` instance. Ballerina does not support function overloading. Therefore, the bindgen tool generates a separate Ballerina function for each overloaded method or constructor. We will improve the function names of the generated bindings in a future release. 
+- All the public instance methods in the `java.io.FileInputStream` class are mapped to methods in the generated Ballerina class. You can refer [the `bindgen` tool](#the-bindgen-tool) section for more details on how other Java class members are mapped into Ballerina bindings.
```

Comment:
```suggestion
- All the public instance methods in the `java.io.FileInputStream` class are mapped to methods in the generated Ballerina class. You can refer [the `bindgen` tool](#the-bindgen-tool) section for more details on how other Java class members are mapped into Ballerina bindings.
```
Please follow the consistent format in adding reference links. See above review comments and fix all occurrences.

Reply:
Done.